### PR TITLE
fix: add buffer, cleanup, reduce size

### PR DESCRIPTION
**Pull request by hugomrdias**
- add buffer related to https://github.com/ipfs/js-ipfs/issues/2924
- remove unnecessary eslint ignore
- remove tweelnacl and use node-forge
- remove browserify-aes  and use node-forge
- use multibase to encode b58
- require only sha256 from multihashing
- reduce bundle size

after all the deps here https://github.com/ipfs/js-ipfs/issues/2924 are merged, libp2p-crypto will be able to be bundle with `node: false` 🎉

### CODE REVIEW

**Comment by jacobheun** [comment]
@hugomrdias have you done any downstream testing of this with libp2p or ipfs?

> reduce bundle size

**Comment by hugomrdias** [comment]
ah yes will update bundle size in aegir.js

155KB -> 122.65KB

Would nice to merge this already with https://github.com/libp2p/js-libp2p-crypto-secp256k1/pull/25 these numbers include this PR.


> @hugomrdias have you done any downstream testing of this with libp2p or ipfs?
not yet but i will im following the path :) i will report here my findings

**Comment by hugomrdias** [comment]
also dont bother fixing the bundle size github check i will do it later.

File: package.json [hunk 1]
```diff
@@ -37,25 +37,20 @@
...
-    "bn.js": "^5.0.0",
-    "browserify-aes": "^1.2.0",
-    "bs58": "^4.0.1",
+    "buffer": "^5.5.0",
     "err-code": "^2.0.0",
     "iso-random-stream": "^1.1.0",
     "keypair": "^1.0.1",
-    "libp2p-crypto-secp256k1": "~0.4.0",
-    "multihashing-async": "~0.8.0",
+    "libp2p-crypto-secp256k1": "libp2p/js-libp2p-crypto-secp256k1#fix/add-buffer",
```
Comment (jacobheun) [inline package.json]: libp2p-crypto-secp256k1 has been released
```suggestion
    "libp2p-crypto-secp256k1": "^0.4.2",
```

**Pull request merged**